src/sql/src/func.rs

```diff
@@ -1974,7 +1976,7 @@ pub static PG_CATALOG_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(|
                 // does not have a limit on identifier length.
             }) => ScalarType::Array(Box::new(ScalarType::String)), 1403;
         },
-        "current_database" => Scalar {
+        "current_database", "current_catalog" => Scalar {
```

Comment:
Nice use of macros! Unfortunately we can't map multiple functions to the same implementation like this because then the functions will share the same OID, that's what the [Cargo test](https://buildkite.com/materialize/test/builds/82307#_) failure is alluding to.

Postgres OIDs are defined [here](https://github.com/postgres/postgres/blob/master/src/include/catalog/pg_proc.dat), but in this case "current_catalog", "current_role", and "user" don't exist in Postgres so we [make up our own OID](https://github.com/MaterializeInc/materialize/blob/main/src/pgrepr-consts/src/oid.rs).

Instead of allowing multiple names for a single function, can you make these all their own function with their own OIDs?

Reply:
Makes sense. Let me know if I got it right.